sleepyq/__init__.py

```diff
@@ -36,66 +36,208 @@ def __init__(self, data):
         super(Sleeper, self).__init__(data)
         self.bed = None
 
+class Light(APIobject):
+    def __init__(self, data):
+        super(Light, self).__init__(data)
+        self.bed = None
+
+class FavSleepNumber(APIobject):
+    def __init__(self, data):
+        super(FavSleepNumber, self).__init__(data)
+        self.bed = None
+        self.left = None
+        self.right = None
+
 class Sleepyq:
     def __init__(self, login, password):
         self._login = login
         self._password = password
         self._session = requests.Session()
+        self._api = "https://api.sleepiq.sleepnumber.com/rest"
+
+    def __makeRequest(self, url, mode="get", data=""):
+        if mode == 'put':
+            r = self._session.put(self._api+url, json=data)
+        else:
+            r = self._session.get(self._api+url)
+        if r.status_code == 401: # HTTP error 401 Unauthorized
+            # Login
+            self.login()
+            # Retry Request
+            if mode == 'put':
```

Comment:
I wonder if it'd be possible to re-use the logic for retrying? Maybe call the method again, and track how many times it was retried?

Reply:
Added ability to count attempts to run requests and limited to 2 tries.  I could make it try a few more times if necessary.